sos/report/plugins/microshift.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins.openshift import Openshift
+
+
+class Microshift(Openshift):
+    """
+    For Microshift users, this plugin does not need to perform any oauth steps
+    of any kind. Users should see the `microshift` plugin enable and NOT the
+    `openshift` plugin during a normal execution of `sos report`. All plugin
+    options are otherwise the same, however users should note that the `token`
+    option is ignored by the Microshift plugin variant.
+    """
+
+    short_desc = 'Microshift Container Platform'
+    plugin_name = 'microshift'
+    packages = ('microshift',)
+    services = ('microshift',)
+    profiles = ()
+    master_localhost_kubeconfig = ''  # TODO: is there a default kubeconfig?
+
+    def _check_oc_logged_in(self):
+        """Microshift does not have the oauth bits that enable the ``oc login``
+        or ``oc whoami`` commands. As such, the ``oc`` commands are otherwise
+        simply available for use. In this case, don't block API collections
+        as the Openshift plugin would when these non-existent commands fail.
+        """
+        return True
+
+    def get_resource_list(self):
+        """
+        Microshift does not support all the same resources as OCP does, so we
+        need to define a shorter list here that only specifies the resources
+        we care about in a microshift environment.
+        """
+        return [
+            'pods',
+            'deployments',
+            # TODO: build this out with the microshift team
+        ]
+
+    def setup(self):
+        super(Microshift, self).setup()
+        # TODO: add microshift-specific collections here that should not be
+        # included in the parent openshift plugin
```

Comment:
There's also some stuff in the parent class that we wouldn't want for MicroShift. We may need a common base class with OCP and MicroShift extensions of it.

For example, there's not a kubelet log on MicroShift. We would want the message about the time it takes to run oc to say "MicroShift" not "OpenShift". We'll want to look at some of the other details, too.

Reply:
Ack on the messaging.

If we specify a collection that doesn't exist on the host, we simply skip it and move on - we won't see an empty `/etc/kubernetes` directory in the archive for example.